Classes/NodeRendering/Dto/DocumentNodeCacheKey.php

```diff
@@ -25,35 +25,42 @@ final class DocumentNodeCacheKey
      */
     protected $dimensions;
 
+    /**
+     * @var string
+     */
+    protected $workspaceName;
+
     /**
      * @var array
      */
     protected $arguments;
 
-    private function __construct(string $nodeIdentifier, array $dimensions, array $arguments)
+    private function __construct(string $nodeIdentifier, array $dimensions, string $workspaceName, array $arguments)
     {
         // we need to make this deterministic
         ksort($arguments);
         ksort($dimensions);
 
         $this->nodeIdentifier = $nodeIdentifier;
         $this->dimensions = $dimensions;
+        $this->workspaceName = $workspaceName;
         $this->arguments = $arguments;
     }
 
 
     public static function fromNodeAndArguments(NodeInterface $node, array $arguments): self
     {
-        return new self($node->getIdentifier(), $node->getContext()->getDimensions(), $arguments);
+        return new self($node->getIdentifier(), $node->getContext()->getDimensions(), $node->getWorkspace()->getName(), $arguments);
     }
 
     public static function fromEnumeratedNode(EnumeratedNode $enumeratedNode)
     {
-        return new self($enumeratedNode->getNodeIdentifier(), $enumeratedNode->getDimensionsFromContextPath(), $enumeratedNode->getArguments());
+        return new self($enumeratedNode->getNodeIdentifier(), $enumeratedNode->getDimensionsFromContextPath(), $enumeratedNode->getWorkspaceNameFromContextPath(), $enumeratedNode->getArguments());
```

Comment:
how does this change do anything? (modifications in this class)

Reply:
This was not fully finished, see TODO below. I think we were lucky so far, that we didn't have collisions due to this.